mps-legacy-sync-plugin/src/main/kotlin/org/modelix/mps/sync/ModelSyncService.kt

```diff
@@ -103,6 +95,23 @@ class ModelSyncService : Disposable, ISyncService {
         }
     }
 
+    private val modelBindings
+        // Use ModelServerConnections.instance.modelServers.rootBindings
+        // instead of ActiveBranch.bindings
+        // as long the not every consumer uses the ModelSyncService to create connections.
+        get() = ModelServerConnections.instance.modelServers
+            .flatMap(ModelServerConnection::rootBindings)
+            .flatMap { rootBinding -> rootBinding.allBindings }
+            .filterIsInstance<ModelBinding>()
+
+    override fun findCloudNodeReference(mpsNode: SNode): List<INode> {
+        return modelBindings.mapNotNull { it.findCloudNodeReference(mpsNode) }
+    }
+
+    override fun findMpsNode(cloudNodeReference: INodeReference): List<SNode> {
+        return modelBindings.mapNotNull { it.findMpsNode(cloudNodeReference) }
```

Comment:
The method is expected to be called quickly after a node is created on the server and then it may not be synchronized to MPS yet. Some `.flush()` calls are probably useful before trying to resolve it.